make.ps1

```diff
@@ -225,20 +186,68 @@ elseif ($command -eq "docs")
 		UtilityNotFound
 	}
 }
-else
-{
-	echo ("Invalid command '{0}'" -f $command)
-}
 
-if ($args.Length -eq 0)
+function FindMSBuild
 {
-	echo "Press enter to continue."
-	while ($true)
+	$msBuildVersions = @("4.0")
+	foreach ($msBuildVersion in $msBuildVersions)
 	{
-		if ([System.Console]::KeyAvailable)
+		$key = "HKLM:\SOFTWARE\Microsoft\MSBuild\ToolsVersions\{0}" -f $msBuildVersion
+		$property = Get-ItemProperty $key -ErrorAction SilentlyContinue
+		if ($property -eq $null -or $property.MSBuildToolsPath -eq $null)
 		{
-			break
+			continue
+		}
+		$path = Join-Path $property.MSBuildToolsPath -ChildPath "MSBuild.exe"
+		if (Test-Path $path)
+		{
+			return $path
 		}
-		Start-Sleep -Milliseconds 50
+	}
+	return $null
+}
+
+
+
```

Comment:
I think we can reduce this three new lines to one, and also add one new line below the `UtilityNotFound` function.